group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -497,6 +499,30 @@ public DeadlineAndEpoch metadataRefreshDeadline() {
         return metadataRefreshDeadline;
     }
 
+    /**
+     * Validates the OffsetCommit request.
+     *
+     * @param memberId          The member id.
+     * @param groupInstanceId   The group instance id.
+     * @param memberEpoch       The member epoch.
+     */
+    @Override
+    public void validateOffsetCommit(
+        String memberId,
+        String groupInstanceId,
+        int memberEpoch
+    ) throws UnknownMemberIdException, StaleMemberEpochException {
+        // When the member epoch is -1, the request comes from either the admin client
+        // or a consumer which does not use the group management facility. In this case,
+        // the request can commit offsets if the group is empty.
+        if (memberEpoch < 0 && members().isEmpty()) return;
```

Comment:
are there cases where members.isEmpty is true but the state is not yet EMPTY?

Reply:
No. EMPTY is actually set based on this condition: https://github.com/apache/kafka/blob/trunk/group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java#L505.

Reply:
I was just curious why isInState(EMPTY) is used in the generic group but not for consumer group, but then I also saw we don't have such a method. 😅

Reply:
Yeah, that's right :).